src/utils/string.ts

```diff
@@ -0,0 +1,31 @@
+import { DonorResponse } from '@/types/donors';
+
+// Perform a set of replacements on a string
+export function replaceMultiple(
+  str: string,
+  replacements: Map<string, string>
+): string {
+  let result: string = str;
+  replacements.forEach((replaceValue, searchValue) => {
+    result = result.replace(searchValue, replaceValue);
+  });
+  return result;
+}
+
+// Replace the email template placeholder values with the corresponding values from a donor and optionally a date
+export function populateEmailTemplate(
```

Comment:
This isn't super "string util" related, and it really is only useful in the context of emails. Can this be moved into `server/actions/mailMerge.ts`?

Reply:
Unfortunately, this prevents it from being used for client-side rendering (e.g. `src/components/email-card/index.tsx`). I could move it to a different file though.